ehcache-api/src/main/java/org/ehcache/spi/resilience/StoreAccessRuntimeException.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi.resilience;
+
+import java.util.concurrent.CompletionException;
+
+/**
+ * A wrapper Runtime exception used when don't want to handle the checkedException an internal operation fails on a {@link org.ehcache.Cache}.
+ *
+ * @author nnares
+ */
+public class StoreAccessRuntimeException extends RuntimeException {
+
+  private static final long serialVersionUID = 6249505400891654776L;
+
+  /**
+   * Creates a new exception wrapping the {@link Throwable cause} passed in.
+   *
+   * @param cause the cause of this exception
+   */
+  public StoreAccessRuntimeException(Throwable cause) {
```

Comment:
My understand was that this type was only needed to transit a StoreAccessException out of a function/lambda. If that is the case then we should constrain these to only be able to be constructed with a `StoreAccessException`. You could then override the `getCause()` method to return this type which might make some of the downstream exception handling simpler.